docs/en/query_language/functions/array_functions.md

```diff
@@ -652,16 +652,36 @@ If you want to get a list of unique items in an array, you can use arrayReduce('
 
 A special function. See the section ["ArrayJoin function"](array_join.md#functions_arrayjoin).
 
-## arrayDifference(arr) {#array_functions-arraydifference}
+## arrayDifference {#arraydifference}
 
-Takes an array, returns an array of differences between adjacent elements. The first element will be 0, the second is the difference between the second and first elements of the original array, etc. The type of elements in the resulting array is determined by the type inference rules for subtraction (e.g. UInt8 - UInt8 = Int16). UInt*/Int*/Float* types are supported (type Decimal is not supported).
+Calculates the difference between adjacent array elements. Returns an array where the first element will be 0, the second is the difference between `a[1] - a[0]`, etc. The type of elements in the resulting array is determined by the type inference rules for subtraction (e.g. `UInt8` - `UInt8` = `Int16`). The type `Decimal` is not supported.
```

Comment:
* "Decimal type is not supported" would sound better. What about other types that have arithmetics like Date/DateTime?
* Link to rules of subtraction + I'd move it to "Returned values" section
* What happens for an empty array?